fission/src/systems/MatchMode.ts

```diff
@@ -0,0 +1,120 @@
+import SimulationSystem from "./simulation/SimulationSystem"
+
+export enum MatchModeType {
+    Sandbox = 0,
+    Autonomous = 1,
+    Teleop = 2,
+    MatchEnded = 3,
+}
+
+class MatchMode {
+    private static instance: MatchMode
+    private matchEnabled: boolean = false
+    private matchModeType: MatchModeType = MatchModeType.Sandbox
+
+    private initialTime: number = 0
+    private timeLeft: number = 0
+    private intervalId: number | null = null
+
+    private constructor() {}
+
+    static getInstance(): MatchMode {
+        if (!MatchMode.instance) {
+            MatchMode.instance = new MatchMode()
+        }
```

Comment:
You should use the nullish coalescing assignment operator here (`??=`), which assigns a variable only if it was initially nullish (ie. undefined or null).